packages/openapi-v3/src/json-to-schema.ts

```diff
@@ -32,9 +37,25 @@ export type SchemaRef = ReferenceObject & {definitions: SchemasObject};
 /**
  * Converts JSON Schemas into a SchemaObject
  * @param json - JSON Schema to convert from
+ * @param visited - A map to keep track of mapped json schemas to handle
+ * circular references
  */
-export function jsonToSchemaObject(json: JsonSchema): SchemaObject | SchemaRef {
-  const result: SchemaObject = {};
+export function jsonToSchemaObject(
+  json: JsonSchema,
+  visited: Map<JsonSchema, SchemaObject | SchemaRef> = new Map(),
+): SchemaObject | SchemaRef {
+  // A flag to check if a schema object is fully converted
+  const converted = 'x-loopback-converted';
```

Comment:
Should we use a private Symbol instead of a string?

Reply:
I thought about that too but decided to use open api extension mechanism to be consistent.